opt_einsum/contract.py

```diff
@@ -674,13 +663,11 @@ def __str__(self):
         return s
 
 
-class _ShapeOnly(np.ndarray):
+def shape_only(shape):
     """Dummy ``numpy.ndarray`` which has a shape only - for generating
     contract expressions.
     """
-
-    def __init__(self, shape):
-        self.shape = shape
+    return np.broadcast_to(np.nan, shape)
```

Comment:
Interesting optimization by NumPy. Simpler to use this, nice change.

Reply:
Yes, subclassing ``ndarray`` I think was having some side-effects when instantiated, whereas this is guaranteed to essentially allocate no memory